pkg/arvo/sys/hoon.hoon

```diff
@@ -910,31 +900,42 @@
     (sub len (met boz dat))
   (swp boz dat)
 ::
-::  Like `rip` but produces n-bit blocks instead of 2^n bit blocks.
-::
-++  ripn
-  ~/  %ripn
-  |=  [bits=@ud x=@]
-  ^-  (list @)
-  ?:  =(0 x)  ~
-  [(end 0 bits x) $(x (rsh 0 bits x))]
-::
 ++  rip                                                 ::  disassemble
   ~/  %rip
-  |=  [=bloq x=@]
+  |=  [a=bloq b=@u c=@]
   ^-  (list @)
-  ?:  =(0 x)  ~
-  [(end bloq 1 x) $(x (rsh bloq 1 x))]
+  ?:  =(0 c)  ~
+  [(end a b c) $(c (rsh a b c))]
 ::
 ++  rsh                                                 ::  right-shift
   ~/  %rsh
   |=  [a=bloq b=@u c=@]
   (div c (bex (mul (bex a) b)))
 ::
+++  run                                                 ::  +turn into atom
+  ~/  %run
+  |=  [a=bloq b=@u c=@ d=$-(@ @)]
+  ^-  @
+  (rep a b (rut a b c d))
+::
+++  rut                                                 ::  +turn into list
+  ~/  %rut
+  |=  [a=bloq b=@u c=@ d=$-(@ @)]
```

Comment:
Any particular reason for this not taking `$-(@ *)` instead? (I don't feel strongly either way, this is a pretty thin wrapper.)

Reply:
now we just need one that round-trips back to a (possibly different-`bloq`) output atom buffer ;)

Reply:
I was focused on `+run`, but this is a good point; there's no reason to restrict `+rut`. It should also be wet, then.